src/jvmMain/kotlin/baaahs/net/JvmNetwork.kt

```diff
@@ -175,6 +182,101 @@ class JvmNetwork : Network {
         }
 
         override val myAddress = IpAddress.mine()
+        override val myHostname = myAddress.address.hostName.replace(Regex("\\.local\\.?$"), "")
+        private val mdns = JvmMdns()
+
+        inner class JvmMdns() : Network.Mdns {
+
+            private val svc = JmDNS.create(InetAddress.getLocalHost())
+
+            override fun register(hostname: String, type: String, proto: String, port: Int, domain: String, params: MutableMap<String, String>): Network.MdnsRegisteredService? {
+                var dom = domain
```

Comment:
You could dedupe with something like:

```kt
val dom = domain.normalizeMdnsDomain()

private fun String.normalizeMdnsDomain(): String {
    var dom = this
    if (dom.startsWith(".")) {
        dom = dom.substring(1)
    }
    if (!dom.endsWith(".")) {
        dom += "."
    }
    return dom
}
```

Reply:
Had to make it public to put it on the `Network.Mdns` interface so I could use it in all the places.